pkg/resources/saml2_integration.go

```diff
@@ -241,21 +392,28 @@ func CreateContextSAML2Integration(ctx context.Context, d *schema.ResourceData,
 		req.WithSaml2SpInitiatedLoginPageLabel(v.(string))
 	}
 
-	if !d.GetRawConfig().AsValueMap()["saml2_enable_sp_initiated"].IsNull() {
-		req.WithSaml2EnableSpInitiated(d.Get("saml2_enable_sp_initiated").(bool))
+	if v := d.Get("saml2_enable_sp_initiated").(string); v != "unknown" {
+		parsed, err := strconv.ParseBool(v)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		req.WithSaml2EnableSpInitiated(parsed)
 	}
 
-	if v, ok := d.GetOk("saml2_snowflake_x509_cert"); ok {
-		req.WithSaml2SnowflakeX509Cert(v.(string))
+	if v := d.Get("saml2_snowflake_x509_cert").(string); v != SnowflakeDefaultStringValuePlaceholder {
+		req.WithSaml2SnowflakeX509Cert(v)
 	}
 
-	if !d.GetRawConfig().AsValueMap()["saml2_sign_request"].IsNull() {
-		req.WithSaml2SignRequest(d.Get("saml2_sign_request").(bool))
+	if v := d.Get("saml2_sign_request").(string); v != "unknown" {
```

Comment:
Why is it sometimes "unknown" and sometimes SnowflakeDefaultStringValuePlaceholder? What's the difference?